lib/luadata.c

```diff
@@ -35,6 +37,7 @@ typedef struct luadata_s {
 	char *ptr;
 	size_t size;
 	uint8_t opt;
+	struct sk_buff *skb;  /* optional sk_buff pointer for packet expansion */
```

Comment:
I think a better approach is to introduce a new option.. e.g., LUADATA_OPT_SKB.. then, we can use container_of to get this struct, right?